packages/ra-core/src/store/localStorageStore.ts

```diff
@@ -116,15 +116,11 @@ export const localStorageStore = (
         },
         reset(): void {
             const storage = getStorage();
-            for (let i = 0; i < storage.length; i++) {
-                if (storage.key(i)?.substring(0, prefixLength) === prefix) {
```

Comment:
The problem was that once you remove an item, the index of the next item decreases... So `storage.key(i)` 'jumps' one item each time we delete one.